stored_requests/backends/http_fetcher/fetcher_test.go

```diff
@@ -61,14 +63,39 @@ func TestReqsAndImps(t *testing.T) {
 	assertErrLength(t, errs, 0)
 }
 
-func TestMissingValues(t *testing.T) {
-	fetcher, close := newEmptyFetcher(t, []string{"req-1", "req-2"}, []string{"imp-1"})
+func TestFetchAccounts(t *testing.T) {
+	fetcher, close := newTestAccountFetcher(t, []string{"acc-1", "acc-2"})
 	defer close()
 
-	reqData, impData, errs := fetcher.FetchRequests(context.Background(), []string{"req-1", "req-2"}, []string{"imp-1"})
-	assertMapKeys(t, reqData)
-	assertMapKeys(t, impData)
-	assertErrLength(t, errs, 3)
+	accData, errs := fetcher.FetchAccounts(context.Background(), []string{"acc-1", "acc-2"})
+	assertMapKeys(t, accData, "acc-1", "acc-2")
+	assertErrLength(t, errs, 0)
+}
+
+func TestFetchAccountsNoData(t *testing.T) {
+	fetcher, close := newFetcherBrokenBackend()
+	defer close()
+	accData, errs := fetcher.FetchAccounts(context.Background(), []string{"req-1"})
+	assertMapKeys(t, accData)
+	assertErrLength(t, errs, 1)
+}
+
+func TestFetchAccount(t *testing.T) {
+	fetcher, close := newTestAccountFetcher(t, []string{"acc-1"})
+	defer close()
+
+	account, errs := fetcher.FetchAccount(context.Background(), "acc-1")
+	assert.Empty(t, errs, "Unexpected error fetching existing account")
+	assert.JSONEq(t, `"acc-1"`, string(account))
+}
+
```

Comment:
Please consider adding a test case for an invalid json response from the http endpoint.

Also, if possible consider updating the tests to a table test format as they are easier to follow and also easier to add/modify/update test cases.

Reply:
Added invalid json test.

Reply:
Regarding the table test: the tests are small and differ in almost every line; the table would have to include all the differences and the eventual custom text. It may not be much of a win considering the code that needs to be added to read and execute the tests will be more dense than any of them.

Reply:
Fair enough! I understand that it will be a lot more work to refactor these tests.